exercises/numbers/primeFactors/primeFactors.js

```diff
@@ -14,22 +14,48 @@
  * @returns {number[]} An array of all the prime factors of the given integer
  */
 function primeFactors(num) {
-  /*
-    Your code goes here.
-
-    Work out one version that works and don't worry about performance.
-
-    If you're having trouble working it out in code, step out of JS-land
-    and use pen/paper, index cards, etc. — anything that helps you think
-    about it without getting stuck in JavaScript syntax.
+  /*pranjal's code
+  for (l = 0; l <= num; l++) {
+    
+    if (l <= 1 ) {
+      return false
+    }
+    for (i = 2; i < l; i++) {
+      if (num%i === 0){
+        return false }
+      
+    }
+    return true
+  }
   */
+  let primearray =[];
+    for (let i=2;i<=num;i++) {
```

Comment:
It's unusual to have `num` changing over the course of the loop. It makes the code very hard to reason about. For example, when does the loop stop?  It's not really when `i === num`, because `num` is modified by the loop.

I'm not totally sure it's correct, I'd have to sit down and reason very carefully about what the code is doing and see if I couldn't find an example input that caused problems.

Instead, try to have your code reflect how you'd do it by hand or how you'd explain it in plain English.

You increment a test divisor (`i`), starting at `2`. Whenever you find an actual divisor, you _know_ it must be prime. So if the first `i` you find is `3`, you divide out all the `3`'s from `num` and then increment the divisor. Once 

If `i` is not a divisor, increment.

Something like:

```js
function primeFactors(num) {
  if (num === 1) {
    return [];
  }

  let factors = [];
  let divisor = 2;

  while (num > 1) {
    if (num % divisor === 0) {
      factors.push(divisor);
      num = num / divisor;
    } else {
      divisor++;
    }

  }

  return factors;
}
```